File: app/src/main/java/com/example/despensacx/ProductoDao.java
package com.example.despensacx;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface ProductoDao {

    @Insert
    long insert(ProductoEntity producto);

    @Update
    void update(ProductoEntity producto);

    @Delete
    void delete(ProductoEntity producto);

    @Query("SELECT * FROM productos WHERE listaId = :listaId")
    LiveData<List<ProductoEntity>> getProductosByLista(long listaId);

    @Query("SELECT * FROM productos WHERE listaId = :listaId")
    List<ProductoEntity> getProductosByListaSync(long listaId);

    @Query("UPDATE productos SET seleccionado = :seleccionado WHERE listaId = :listaId")
    void setAllSeleccionadoSync(long listaId, boolean seleccionado);

    @Query("SELECT * FROM productos")
    List<ProductoEntity> getAllSync();
}

File: app/src/main/kotlin/com/example/despensacx/data/ProductoDao.kt
package com.example.despensacx.data

import androidx.lifecycle.LiveData
import androidx.room.*

@Dao
interface ProductoDao {
    @Insert
    suspend fun insert(producto: ProductoEntity): Long

    @Update
    suspend fun update(producto: ProductoEntity)

    @Delete
    suspend fun delete(producto: ProductoEntity)

    @Query("SELECT * FROM productos WHERE listaId = :listaId")
    fun getProductosByLista(listaId: Long): LiveData<List<ProductoEntity>>

    @Query("SELECT * FROM productos WHERE listaId = :listaId")
    fun getProductosByListaSync(listaId: Long): List<ProductoEntity>

    @Query("UPDATE productos SET seleccionado = :seleccionado WHERE listaId = :listaId")
    suspend fun setAllSeleccionadoSync(listaId: Long, seleccionado: Boolean)

    @Query("SELECT * FROM productos")
    fun getAllSync(): List<ProductoEntity>

    @Query("DELETE FROM productos")
    suspend fun deleteAll()
}
